app/views/catalog/_marc_search_results_document_fields.html.erb

```diff
@@ -17,23 +17,15 @@
       <% end if uniform_title[:unmatched_vernacular].present? %>
     <% end %>
 
-    <% author_creator = link_to_author_from_marc(document.to_marc) %>
-    <% unless author_creator.blank? %>
-      <li><%= render_field_list_from_marc(author_creator) %></li>
-    <% end %>
-
-    <% author_corp = link_to_data_with_label(document, "Corporate Author", 'author_corp_display', {controller: 'catalog', action: 'index', search_field: 'search_author'}) %>
-    <% unless author_corp.blank? %>
-      <%= render "field_list_from_index", :fields => author_corp %>
-    <% end %>
-
-    <% author_meeting = link_to_data_with_label(document, "Meeting", 'author_meeting_display', {controller: 'catalog', action: 'index', search_field: 'search_author'}) %>
-    <% unless author_meeting.nil? %>
-      <%= render "field_list_from_index", :fields => author_meeting %>
+    <% %i[creator corporate_author meeting].each do |target| %>
+      <% linked_author = LinkedAuthor.new(document, target) %>
```

Comment:
What would you think about adding accessors for the different author/creator flavors in https://github.com/sul-dlss/SearchWorks/blob/master/app/models/concerns/marc_metadata.rb?

This would allow us to avoid referencing classes directly in our views.